js/qmwp-add-measurement.js

```diff
@@ -0,0 +1,587 @@
+            var that = this,
+                currentCategory = '';
+            jQuery.each(items, function (index, item) {
+                if (item.category != currentCategory) {
+                    ul.append('<li class="ui-autocomplete-category">' + item.category + '</li>');
+                    currentCategory = item.category;
+                }
+                that._renderItemData(ul, item);
+            });
+
+        }
+    });
+
+    var request = {message: 'getVariables', params: {}};
+
+    Quantimodo.getVariables(null, function (vars) {
+
+        variables = vars;
+        var varnames = [];
+        var categories = [];
+        variableCategorySelect = document.getElementById('addmeasurement-variable-category');
+
+        if (variables.length) {
+            jQuery.each(variables.sort(function (a, b) {
+                return a.name.localeCompare(b.name);
+            }), function (_, variable) {
+                varnames.push(variable.name);
+            });
+        }
+
+    });
+
+};
+
+var loadVariableUnits = function () {
+
+    Quantimodo.getUnits(null, function (variableUnits) {
+        units = variableUnits;
+        unitSelect = document.getElementById('addmeasurement-variable-unit');
+
+        if (units.length) {
+            jQuery.each(units.sort(function (a, b) {
+                return a.name.localeCompare(b.name);
+            }), function (_, unit) {
+                unitSelect.options[unitSelect.options.length] = new Option(unit.name, unit.abbreviatedName);
+            });
+        }
+    });
+
+};
+
+// Load option for the Distance
+var loadAddVariableUnits = function () {
+
+    Quantimodo.getUnits(null, function (variableUnits) {
+        units = variableUnits;
+        unitSelect = document.getElementById('add-addmeasurement-variable-unit');
+
+        if (units.length) {
+            jQuery.each(units.sort(function (a, b) {
+                return a.name.localeCompare(b.name);
+            }), function (_, unit) {
+                unitSelect.options[unitSelect.options.length] = new Option(unit.name, unit.abbreviatedName);
+            });
+        }
+    });
+
+};
+
+function addZero(i) {
+    if (i < 10) {
+        i = '0' + i;
+    }
+    return i;
+}
+
+var loadDateTime = function () {
+
+    jQuery('#addmeasurement-variable-date').datetimepicker({
+        dayOfWeekStart: 1,
+        lang: 'en',
+        startDate: '1986/01/05',
+        format: 'h:i A m/d/Y'
+    });
+
+
+    var currentTime = new Date();
+
+    var j_years = currentTime.getFullYear();
+    var j_months = currentTime.getMonth();
+    var j_date = currentTime.getDate();
+
+    var j_hours = addZero(currentTime.getHours());
+    var j_minutes = addZero(currentTime.getMinutes());
+
+    var jjj_minutes = ((currentTime.getHours() % 12) ? currentTime.getHours() % 12 : 12) + ':' + currentTime.getMinutes() + (currentTime.getHours() < 12 ? 'AM' : 'PM');
+
+    var c_date = j_months + '/' + j_date + '/' + j_years;
```

Comment:
Identifier 'c_date' is not in camel case.<br>Identifier 'j_months' is not in camel case.<br>Identifier 'j_date' is not in camel case.<br>Identifier 'j_years' is not in camel case.<br>Too many errors. (66% scanned).